sourcebundle/builder.go

```diff
@@ -73,12 +73,16 @@ type Builder struct {
 	// selected version of each module registry package.
 	resolvedRegistry map[registryPackageVersion]sourceaddrs.RemoteSource
 
+	// resolvedRegistryVersionDeprecations tracks potential deprecations for
+	// each package version
+	packageVersionDeprecations map[registryPackageVersion]*RegistryVersionDeprecation
+
 	// registryPackageVersions caches responses from module registry calls to
 	// look up the available versions for a particular module package. Although
 	// these could potentially change while we're running, we assume that the
 	// lifetime of a particular Builder is short enough for that not to
 	// matter.
```

Comment:
Might be worth noting here that this cache includes the deprecation data, as well as the package versions.

Reply:
Included 👍 https://github.com/hashicorp/go-slug/pull/59/commits/be5d90781af6a49d0b5f5dc5fca24e185629c14a